### PR TITLE
rustc: 1.69.0 -> 1.70.0

### PR DESCRIPTION
**Pull request by figsoda**
https://github.com/rust-lang/rust/releases/tag/1.70.0

built fd and vector on {aarch64,x86_64}-linux on master

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zowoq** [comment]
Can revert 9861cf4a7f5f39f82878c7669232eb68eadf7f78 as sparse is now the default.

**Review by zowoq — APPROVED** [review]
Built passthru.tests on x86_64-darwin and firefox on x86_64-linux.

**Comment by trofi** [comment]
Bisect claims that ff16966860a59e8b8db805bf57d41f4deabdc395 `rustc: 1.69.0 -> 1.70.0` broke `evcxr` in `staging`:

```
$ nix build --no-link -f. -L evcxr
...
evcxr> failures:
evcxr> ---- partially_inferred_variable_type stdout ----
evcxr> ======== last src ========
evcxr> #![allow(unused_imports, unused_mut, dead_code)]
evcxr>             struct EvcxrUserCodeError {}
evcxr>             impl<T: std::fmt::Display> From<T> for EvcxrUserCodeError {
evcxr>                 fn from(error: T) -> Self {
evcxr>                     eprintln!("{}", error);
evcxr>                     EvcxrUserCodeError {}
evcxr>                 }
evcxr>             }
evcxr> fn evcxr_variable_store<T: 'static>(_: T) {}
evcxr> #[allow(unused_variables)]
evcxr> async fn evcxr_analysis_wrapper(
evcxr> ) -> Result<(), EvcxrUserCodeError> {
evcxr> let v : Vec < _ > = (1 .. 10).collect() ;
evcxr> evcxr_variable_store(v);
evcxr> Ok(())
evcxr> }
evcxr> ==========================
evcxr> Couldn't automatically determine type of variable `v`.
evcxr> Please give it an explicit type.
evcxr> thread 'partially_inferred_variable_type' panicked at 'Unexpected compilation error. See above for details', evcxr/tests/integration_tests.rs:677:5
evcxr> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
evcxr> failures:
evcxr>     partially_inferred_variable_type
evcxr> test result: FAILED. 35 passed; 1 failed; 0 ignored; 0 measured; 1 filtered out; finished in 42.87s
```

**Comment by figsoda** [comment]
#236140

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -16339,20 +16339,20 @@ with pkgs;
     inherit (darwin) apple_sdk;
   };
 
-  rust_1_69 = callPackage ../development/compilers/rust/1_69.nix {
+  rust_1_70 = callPackage ../development/compilers/rust/1_70.nix {
     inherit (darwin.apple_sdk.frameworks) CoreFoundation Security SystemConfiguration;
     llvm_15 = llvmPackages_15.libllvm;
```
Comment (fgaz) [inline pkgs/top-level/all-packages.nix:16344]: llvm should be updated to 16, see https://github.com/rust-lang/rust/releases/tag/1.70.0 (internal changes)

**Pull request merged**